### DISCUSSION TITLE
TreeListView in Avalonia

### DISCUSSION THREAD

**Discussion by ShrutiJaiswal1494** [AvaloniaUI/Avalonia#5585] [Q&A] [answered]
I need to create a TreeView that shows data like a DataGrid such that I have a TreeView at left side with columns at right side. I see in WPF we can achieve this using a TreeListView. Is something like this available for avalonia?

**Comment by Gramli** [comment]
Did you tried create custom TreeDataTemplate?

Something like this:

```xml
  <TreeView Items="{Binding Items}">
        <TreeView.DataTemplates>
        <TreeDataTemplate ItemsSource="{Binding SubItems}">
          <Grid>
            <Grid.ColumnDefinitions>
                    <ColumnDefinition Width="*"/>
                    <ColumnDefinition Width="*"/>
            </Grid.ColumnDefinitions>
                <TextBlock Text="{Binding Name}" Grid.Column="0"/>
                <TextBlock Text="{Binding Text}" Grid.Column="1"/>
            </Grid>
        </TreeDataTemplate>
    </TreeView.DataTemplates>
</TreeView>
```

And then result is:
![image](https://user-images.githubusercontent.com/11515648/110383061-14d85300-805c-11eb-932c-b4a6ab751a8e.png)

**Reply by ShrutiJaiswal1494** [reply]
Thanks for the response. I am currently using this structure only but the problem is the entire row shifts to left for each level.  However I do not want the values on the second column to shift with each level and should appear in a single line (something like the image below). Is there any workaround for this?

![image](https://user-images.githubusercontent.com/79670853/110389356-43810a00-8019-11eb-9e8d-640675f04051.png)

**Comment by Gramli** [comment, marked as answer]
Hope I understand, you can set Column.Width, then the second, third .. columns will be aligned.

```xml
      <TreeView Items="{Binding Items}">
        <TreeView.DataTemplates>
          <TreeDataTemplate DataType="models:Item" ItemsSource="{Binding SubItems}">
            <Grid>
              <Grid.ColumnDefinitions>
                <ColumnDefinition Width="*"/>
                <ColumnDefinition Width="200"/>
                <ColumnDefinition Width="200"/>
              </Grid.ColumnDefinitions>
              <TextBlock Text="{Binding Name}" Grid.Column="0"/>
              <TextBlock Text="{Binding Type}" Grid.Column="1"/>
              <TextBlock Text="{Binding Data}" Grid.Column="2"/>
            </Grid>
          </TreeDataTemplate>
        </TreeView.DataTemplates>
  </TreeView>
```
![image](https://user-images.githubusercontent.com/11515648/110459430-43424680-80cd-11eb-991d-bb3080d9978b.png)
